ddtrace/ext/git.py

```diff
@@ -95,6 +102,34 @@ def extract_user_info(cwd=None):
     }
 
 
+def extract_git_version(cwd=None):
+    output = _git_subprocess_cmd("--version")
+    version_info = tuple([int(part) for part in output.split()[-1].split(".")])
+    return version_info
+
+
+def extract_remote_url(cwd=None):
+    remote_url = _git_subprocess_cmd("config --get remote.origin.url", cwd=cwd)
+    return remote_url
+
+
+def extract_latest_commits(cwd=None):
+    latest_commits = _git_subprocess_cmd(["log", "--format=%H", "-n", "1000" '--since="1 month ago"'], cwd=cwd)
+    return latest_commits.split("\n") if latest_commits else []
+
+
+def get_rev_list_excluding_commits(commit_shas, cwd=None):
+    command = ["rev-list", "--objects", "--filter=blob:none"]
+    if extract_git_version(cwd=cwd) >= (2, 23, 0):
```

Comment:
Yes

Reply:
this might be an issue for large repos. If `--since` and `--no-object-names` are not supported for <2.23 we might settle with not being compatible with older versions of git. 2.23 seems to have been released more than 4 years ago (https://github.com/git/git/blob/master/Documentation/RelNotes/2.23.0.txt) so I think it's fine. 

I don't think we need to do that in this PR though